bin/Makefile

```diff
@@ -17,7 +17,7 @@
 all: s2nc s2nd
 include ../s2n.mk
 
-LDFLAGS += -L../lib/ -L${LIBCRYPTO_ROOT}/lib -ls2n ${LIBS} ${CRYPTO_LIBS}
+LDFLAGS += -L../lib/ -L${LIBCRYPTO_ROOT}/lib -Wl,-Bstatic -ls2n -Wl,-Bdynamic ${CRYPTO_LIBS} ${LIBS}
```

Comment:
Isn't that stuff currently determined based on the compilation mode of s2n itself?

Reply:
This is instructing the linker to use the static library.  we compile both, in both CMake (static only by default) and Makefile builds.  The reason for static on bin is that we call some internal APIs from these sample applications, and I'd rather not have S2N_API littered throughout header files (the one exception to this is in the s2n_tls13.h in this PR).

Reply:
Understood. I just thought cmake sets these kinds of flags automatically, and based on the settings you do in cmake directly (e.g. via `target_link_libraries`).  So as long as something is expressible in cmake one shouldn't do it with explicit arguments, since it might conflict with what cmake is doing and/or break cross-platform builds.

In this specific case I am however not exactly sure what is set by cmake. I thought we had the 2 options for building a statically and dynamically linked version of s2n. And depending on that mode we also would have set static/dynamic linking to libcrypto, which you set explicitly to dynamic here.

Reply:
I see your point now - I'll just specify the full path to the .a files to make this explict for libs2n.a and libtests2n.a